### PR TITLE
[Feature #19437] Add marking and sweeping time to GC.stat

### PR DESCRIPTION
**Pull request by peterzhu2118**
There is a `time` key in GC.stat that gives us the total time spent in GC. However, we don't know what proportion of the time is spent between marking and sweeping. This makes it difficult to tune the GC as we're not sure where to focus our efforts on.

This PR adds keys `marking_time` and `sweeping_time` to GC.stat for the time spent marking and sweeping, in milliseconds.

cc. @byroot

### CODE REVIEW

File: gc.c [hunk 1]
```diff
@@ -11300,7 +11370,9 @@ gc_stat_internal(VALUE hash_or_sym)
         rb_hash_aset(hash, gc_stat_symbols[gc_stat_sym_##name], SIZET2NUM(attr));
 
     SET(count, objspace->profile.count);
-    SET(time, (size_t) (objspace->profile.total_time_ns / (1000 * 1000) /* ns -> ms */)); // TODO: UINT64T2NUM
+    SET(time, (size_t)ns_to_ms(objspace->profile.total_time_ns)); // TODO: UINT64T2NUM
+    SET(marking_time, (size_t)ns_to_ms(objspace->profile.marking_time_ns));
+    SET(sweeping_time, (size_t)ns_to_ms(objspace->profile.sweeping_time_ns));
```
Comment (byroot) [inline gc.c:11375]: Hum, I haven't looked at the patch in close detail yet, but couldn't `profile.total_time_ns` be eliminated?

Intuitively I'd think we could compute it on the fly with `marking_time_ns + sweeping_time_ns`? That would be one less counter to increment.

Comment (peterzhu2118) [reply gc.c:11375]: `total_time_ns` also includes time spent on finalizers, but I guess if we added a `finalizing_time_ns` then `total_time_ns == marking_time_ns + sweeping_time_ns + finalizing_time_ns`.

Comment (byroot) [reply gc.c:11375]: Couldn't we consider `finalizing` time as part of sweeping?

Comment (peterzhu2118) [reply gc.c:11375]: Oh never mind, it looks like `gc_enter_event_measure_p` returns false for finalizing, so the time doesn't get measured. So you're right, `total_time_ns == marking_time_ns + sweeping_time_ns`.

**Pull request merged**